src/window-rules.c

```diff
@@ -135,3 +145,21 @@ window_rules_get_property(struct view *view, const char *property)
 	}
 	return LAB_PROP_UNSPECIFIED;
 }
+
+bool
+window_rules_get_custom_border_color(struct view *view, float *rgba)
+{
+	struct window_rule *rule;
+	wl_list_for_each_reverse(rule, &rc.window_rules, link) {
+		if (view_matches_criteria(rule, view)) {
+			if (rule->has_custom_border) {
+				memcpy(rgba, rule->custom_border_color, sizeof(float)*4);
+				wlr_log(WLR_DEBUG,
+					"Custom color was found in window rules list: %f, %f, %f, %f\n",
+					rgba[0], rgba[1], rgba[2], rgba[3]);
+				return true;
+			}
+		}
+	}
+	return false;
+}
```

Comment:
1. I don't think the debug info is particularly useful here.
2. This function should only take the `struct view` argument and return a float pointer to the internal `rule->custom_border_color` (if the custom color has been defined) or `NULL`. There's no value in copying the values around, and using the pointer simplifies logic elsewhere.